### ISSUE TITLE
java.lang.IllegalArgumentException in JobIntentService on Android 8.1

### ISSUE THREAD

**Issue by mvisionapps** [OneSignal/OneSignal-Android-SDK#649]
I updated my app to use SDK 3.10.2. I'm seeing some issues on Android 8.1 and 8.0 on different devices  with the following exception
```java
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3208)
  at android.app.ActivityThread.-wrap17 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1684)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:172)
  at android.app.ActivityThread.main (ActivityThread.java:6590)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:438)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:807)
Caused by: java.lang.IllegalStateException: 
  at android.app.ContextImpl.startServiceCommon (ContextImpl.java:1521)
  at android.app.ContextImpl.startService (ContextImpl.java:1477)
  at android.content.ContextWrapper.startService (ContextWrapper.java:650)
  at com.onesignal.JobIntentService$CompatWorkEnqueuer.enqueueWork (JobIntentService.java:204)
  at com.onesignal.JobIntentService.enqueueWork (JobIntentService.java:534)
  at com.onesignal.q.a (NotificationBundleProcessor.java:465)
  at com.onesignal.q.a (NotificationBundleProcessor.java:430)
  at com.onesignal.GcmBroadcastReceiver.a (GcmBroadcastReceiver.java:125)
  at com.onesignal.GcmBroadcastReceiver.onReceive (GcmBroadcastReceiver.java:72)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:3201)
```
Already included these line in manifest 
```xml
<service
   android:name=".util.NotificationExtender"
   android:permission="android.permission.BIND_JOB_SERVICE"
   android:exported="false">
   <intent-filter>
       <action android:name="com.onesignal.NotificationExtender" />
   </intent-filter>
</service>
```
@jkasten2

**Comment by jkasten2** [comment]
@mvisionapps Thanks for the stacktrace, are you able to reproduce this issue on your own devices? Or only reports from the Google Play Console? If you can reproduce the issue on your own devices there should be a more detailed message just after `java.lang.RuntimeException: ` that would be helpful.

If you can reproduce the issue can you let us know which device models are being reported as crashing?

Thanks.

**Comment by mvisionapps** [comment]
@jkasten2 Should we wait for next update or Is there any fix for the issue? I cannot find any solution  in the referenced issue #498

**Comment by bhaveshrangani** [comment]
Same issue after i updated to 3.10.2. have you found any solution?

**Comment by jkasten2** [comment]
@gsksarath Sorry for the delay, we just released [version 3.10.3](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.10.3) with a fix for this. It will take maven central a few hours to sync the update so you might not see it right away.
